src/test/scala/chiselTests/PrefixSpec.scala

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.experimental.hierarchy.{instantiable, public, Definition, Instance, Instantiate}
+import circt.stage.ChiselStage.emitCHIRRTL
+
+class PrefixSpec extends ChiselFlatSpec with ChiselRunners with Utils with MatchesAndOmits {
+
+  behavior.of("withPrefix")
+
+  it should "prefix modules in a withModulePrefix block, but not outside" in {
+    class Foo extends RawModule {
+      val a = Wire(Bool())
+    }
+
+    class Top extends RawModule {
+      val foo = Module(new Foo)
+
+      withModulePrefix("Pref") {
+        val pref_foo = Module(new Foo)
+      }
+    }
+
+    val chirrtl = emitCHIRRTL(new Top)
+
+    val lines = """
+      {
+        "class":"chisel3.ModulePrefixAnnotation",
+        "target":"~Top|Pref_Foo",
+        "prefix":"Pref_"
+      }
+
+      module Foo :
+        wire a : UInt<1>
+      module Pref_Foo :
+        wire a : UInt<1>
+      module Top :
+        inst foo of Foo
+        inst pref_foo of Pref_Foo
+        """.linesIterator.map(_.trim).toSeq
```

Comment:
```suggestion
     |"class":"chisel3.ModulePrefixAnnotation",
     |"target":"~Top|Pref_Foo",
     |"prefix":"Pref_"
     |module Foo :
     |module Pref_Foo :
     |module Top :
     |  inst foo of Foo
     |  inst pref_foo of Pref_Foo
     |""".stripMargin.lines
```
This is a very minor nit, but you could make this a little more concise (and make it more obvious what we're actually trying to check), no need to check for the `wire a` nor really the `{` or `}` of the annotation. Also `.stripMargin` probably would obviate the need for the `trim`. That being said, I didn't know about `linesIterator` (or `lines`), good find!